service/history/workflow/util.go

```diff
@@ -272,12 +288,11 @@ func GetActiveDomainEntry(
 	domainUUID string,
```

Comment:
Can this move to shard context?

Reply:
+1
This would makes more sense as it does not have anything workflow related.

Reply:
Yeah, I was thinking probably this should be moved to `domain` package instead of `shard` as it has not much to do with shard, which is only for getting the domain cache.

Reply:
Synced with @yux0 offline. So `GetActivityDomainEntry` now is a method of the `DomainCache` interface and `ValidateDomainUUID` is moved to `common` package.